tests/integrationv2/test_pq_handshake.py

```diff
@@ -253,6 +268,94 @@ def test_s2nc_to_s2nd_pq_handshake(managed_process, protocol, certificate, clien
         assert_s2n_negotiation_parameters(results, expected_result)
 
 
+@pytest.mark.parametrize("s2n_client_policy", [Ciphers.PQ_TLS_1_3_2023_06_01], ids=get_parameter_name)
+@pytest.mark.parametrize("awslc_server_group", [KemGroups.SecP256r1Kyber768Draft00, KemGroups.X25519Kyber768Draft00], ids=get_parameter_name)
+def test_s2nc_to_awslc_pq_handshake(managed_process, s2n_client_policy, awslc_server_group):
+
+    if not pq_enabled():
+        pytest.skip("PQ not enabled")
+
+    if "awslc" not in get_flag(S2N_PROVIDER_VERSION):
+        pytest.skip("s2n must be compiled with awslc libcrypto in order to test PQ TLS compatibility")
+
+    if "fips" in get_flag(S2N_PROVIDER_VERSION):
+        pytest.skip("No FIPS validated version of AWS-LC has support for negotiating Hybrid PQ TLS yet")
+
+    port = next(available_ports)
+
+    awslc_env_vars = dict()
+    awslc_env_vars["PATH"] = os.path.abspath("../../test-deps/awslc/bin")
+    awslc_env_vars["LD_LIBRARY_PATH"] = os.path.abspath("../../test-deps/awslc/lib")
+
+    s2nc_client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        port=port,
+        insecure=True,
+        cipher=s2n_client_policy,
+        protocol=Protocols.TLS13)
+
+    awslc_server_options = ProviderOptions(
+        mode=Provider.ServerMode,
+        port=port,
+        protocol=Protocols.TLS13,
+        env_overrides=awslc_env_vars,
+        curve=Curves.from_name(awslc_server_group.oqs_name))
+
+    awslc_server = managed_process(BoringSSL, awslc_server_options, timeout=5)
+    s2n_client = managed_process(S2N, s2nc_client_options, timeout=5)
+    expected_result = EXPECTED_RESULTS.get((s2n_client_policy, awslc_server_group), None)
+
+    awslc_result = next(awslc_server.get_results())
+    assert_awslc_negotiation_parameters(awslc_result, expected_result)
+
+    s2nd_result = next(s2n_client.get_results())
+    assert_s2n_negotiation_parameters(s2nd_result, expected_result)
+
+
+@pytest.mark.parametrize("s2n_server_policy", [Ciphers.PQ_TLS_1_3_2023_06_01], ids=get_parameter_name)
+@pytest.mark.parametrize("awslc_client_group", [KemGroups.SecP256r1Kyber768Draft00, KemGroups.X25519Kyber768Draft00], ids=get_parameter_name)
+def test_s2nd_to_awslc_pq_handshake(managed_process, s2n_server_policy, awslc_client_group):
+
+    if not pq_enabled():
+        pytest.skip("PQ not enabled")
+
+    if "awslc" not in get_flag(S2N_PROVIDER_VERSION):
+        pytest.skip("s2n must be compiled with awslc libcrypto in order to test PQ TLS compatibility")
+
+    if "fips" in get_flag(S2N_PROVIDER_VERSION):
+        pytest.skip("No FIPS validated version of AWS-LC has support for negotiating Hybrid PQ TLS yet")
+
+    port = next(available_ports)
+
+    awslc_env_vars = dict()
+    awslc_env_vars["PATH"] = os.path.abspath("../../test-deps/awslc/bin")
```

Comment:
It feels a bit odd to me that we have to manually specify these. We currently have AWS-LC integ tests, and as far as I can tell we don't have to specify these for the other tests?

Ideally we'd launch aws-lc here the same way that we do in other tests. If that's not possible, could you add a comment explaning why the extra env variables are needed?

Reply:
>  as far as I can tell we don't have to specify these for the other tests?

[We do still set up environment variables when testing s2n with libOQS's fork of Openssl](https://github.com/aws/s2n-tls/blob/a9a07a25fa0d897ba6ad2b161b4b8ea9e97b6abd/tests/integrationv2/test_pq_handshake.py#L313). 

Removing all `awslc_env_vars` resulted in CI failure, from trial and error it seems that setting `awslc_env_vars["PATH"] = os.path.abspath("../../test-deps/awslc/bin")` is the only required value for this to pass in CI.

Reply:
Ideally we wouldn't be using any CI specific paths in our integration tests ("test-deps") since it makes these tests really hard to run from a local dev machine. _however_ I've found that our integration tests are already really difficult to run from a local dev machine, so it does feel a little rich to hold you to that standard 😉.